Bogus error when using `..` in struct expr with all fields present, in a const ("error[E0016]: blocks in constants are limited to items and tail expressions")

```rust
struct X { a: i32, b: i32 }

#[allow(unused_variables)]
fn main() {
    const DX: X = X { a: 0, b: 0 };
    const X1: X = X { a: 1, ..DX };  // ok
    let   x2    = X { a: 1, b: 2, ..DX };  // ok
    const X3: X = X { a: 1, b: 2, ..DX };  // error[E0016]
}
```

* [ ] It's weird that `X1` and `x2` are allowed, but not `X3`.
* [ ] Even if that's intentional, the error message seems off.

--- a/src/test/ui/consts/issue-39161-bogus-error.rs
+++ b/src/test/ui/consts/issue-39161-bogus-error.rs
@@ -1,0 +1,13 @@
+// check-pass
+
+pub struct X {
+    pub a: i32,
+    pub b: i32,
+}
+
+fn main() {
+    const DX: X = X { a: 0, b: 0 };
+    const _X1: X = X { a: 1, ..DX };
+    let _x2 = X { a: 1, b: 2, ..DX };
+    const _X3: X = X { a: 1, b: 2, ..DX };
+}

